src/data_generators/data_set_parameters.py

```diff
@@ -0,0 +1,57 @@
+# Copyright 2021 The Private Cardinality Estimation Framework Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+"""Defines the parameters for generating a DataDesign."""
+
+from numpy.random import Generator
+from typing import NamedTuple
+from typing import Type
+from typing import Any
+
+
+class GeneratorParameters(NamedTuple):
+  """Parameters to create one PricingGenerator.
+
+    generator: Class of the generator (e.g. HomogeneousImpressionGenerator)
+    params : Parameters dict used to initialize that class.
+    """
+  #  TODO(uakyol): create a Generator parent class for impression, pricing and
+  #  overlap generators and change this type.
+  generator: Type[Any]
+  params: dict
+
+
+class DataSetParameters(NamedTuple):
+  """Parameters to create one DataSet.
+
+  Does not include randomness
+
+    num_reps:  Number of repetitions for this DataSet to be randomly generated.
+    num_publishers: Number of publishers in this DataSet.
+    largest_publisher_size: Maximum possible reach of the largest publisher.
+    publisher_size_decay_rate: Decay rate for calculating publisher sizes from
```

Comment:
As discussed in our meeting, this will become "ratio_of_largest_to_smallest".

Reply:
done